server/service/src/invoice/outbound_shipment/update/generate.rs

```diff
@@ -53,6 +58,20 @@ pub(crate) fn generate(
         None
     };
 
+    let location_movements = if let Some(batches) = batches_to_update.clone() {
+        let generate_movement = batches
+            .iter()
+            .filter_map(|batch| match batch.location_id {
+                Some(_) => Some(generate_exit_location_movement(batch, store.to_string())),
```

Comment:
I think that this should be extended.. and that we should only record an `exit_datetime` for existing `location_movement` rows, and only if the stock level is reduced to 0 by the outbound shipment.

because the location_movement isn't recording quantity, or directly matching the shipment to the movement we don't know which specific units of stock were affected.